okapi-core/src/main/java/org/folio/okapi/managers/DiscoveryManager.java

```diff
@@ -578,23 +587,28 @@ void health(DeploymentDescriptor md,
       fut.handle(new Success<>(hd));
       return;
     }
-    httpClient.get(new RequestOptions().setAbsoluteURI(url), res1 -> {
-      if (res1.failed()) {
-        hd.setHealthMessage("Fail: " + res1.cause().getMessage());
-        hd.setHealthStatus(false);
-        fut.handle(new Success<>(hd));
+    httpClient.request(new RequestOptions().setAbsoluteURI(url).setMethod(HttpMethod.GET), req -> {
+      if (req.failed()) {
+        fail(req.cause(), hd, fut);
         return;
       }
-      HttpClientResponse response = res1.result();
-      response.endHandler(res -> {
-        hd.setHealthMessage("OK");
-        hd.setHealthStatus(true);
-        fut.handle(new Success<>(hd));
-      });
-      response.exceptionHandler(res -> {
-        hd.setHealthMessage("Fail: " + res.getMessage());
-        hd.setHealthStatus(false);
-        fut.handle(new Success<>(hd));
+      req.result().end();
+      req.result().onComplete(res -> {
+        if (res.failed()) {
+          fail(res.cause(), hd, fut);
+          return;
+        }
+        HttpClientResponse response = res.result();
+        response.endHandler(x -> {
+          hd.setHealthMessage("OK");
+          hd.setHealthStatus(true);
+          fut.handle(new Success<>(hd));
+        });
+        response.exceptionHandler(x -> {
+          hd.setHealthMessage("Fail: " + x.getMessage());
+          hd.setHealthStatus(false);
+          fut.handle(new Success<>(hd));
```

Comment:
x should be renamed to e and the fail method been used.

Reply:
ok